File: code-coaches/hard/new-drivers-license/NewDriversLicense.java
import java.util.*;

class NewDriversLicense {

    public static int getWaitingTime(String myName, int numAgents, List<String> customers) {
        int timeForLicense = 20;
        int positionInQueue = customers.indexOf(myName) + 1;
        int waitingTime = (int) Math.ceil((double) positionInQueue / numAgents) * timeForLicense;
        return waitingTime;
    }

    public static void main(String[] args) {
        Scanner scanner = new Scanner(System.in);

        String myName = scanner.nextLine();
        int numAgents = Integer.parseInt(scanner.nextLine());
        List<String> customers = new ArrayList<>(Arrays.asList(scanner.nextLine().split(" ")));

        customers.add(myName);
        Collections.sort(customers);
        System.out.println(getWaitingTime(myName, numAgents, customers));

        scanner.close();
    }
}

File: code-coaches/hard/new-drivers-license/new_drivers_license.py
from math import ceil


def get_waiting_time(my_name: str, num_agents: int, customers: list[str]) -> int:
    time_for_license = 20
    position_in_queue = ceil((customers.index(my_name) + 1) / num_agents)
    waiting_time = position_in_queue * time_for_license
    return waiting_time


def main():
    my_name = input()
    num_agents = int(input())
    customers = input().split()

    customers.append(my_name)
    customers.sort()
    print(get_waiting_time(my_name, num_agents, customers))


if __name__ == "__main__":
    main()
